xbmc/input/keyboard/XkbcommonKeymap.cpp

```diff
@@ -0,0 +1,348 @@
+/*
+ *      Copyright (C) 2017 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#if defined(HAVE_XKBCOMMON)
```

Comment:
Yes, using CORE_PLATFORM_NAME for conditional include now. My only gripe is that if users set CORE_PLATFORM_NAME to e.g. `Wayland` (with a capital `W`), it won't compile since CMake string comparisons are case-sensitive. But maybe I should submit a PR for normalizing the variable in the main Kodi repo.